pandas/core/groupby/groupby.py

```diff
@@ -3272,7 +3272,10 @@ def first(x: Series):
                 """Helper function for first item that isn't NA."""
                 arr = x.array[notna(x.array)]
                 if not len(arr):
-                    return np.nan
```

Comment:
could do `x.array._na_value`?  avoids a copy here

Reply:
One better...I can revert to original Pandas code.